algosdk/source_map.py

```diff
@@ -0,0 +1,102 @@
+from typing import Dict, Any, List, Tuple
+
+from algosdk.error import SourceMapVersionError
+
+
+class SourceMap:
+    """
+    Decodes a VLQ-encoded source mapping between PC values and TEAL source code lines.
+    Spec available here: https://sourcemaps.info/spec.html
+
+    Args:
+        source_map (dict(str, Any)): source map JSON from algod
+    """
+
+    def __init__(self, source_map: Dict[str, Any]):
+
+        self.version: int = source_map["version"]
+
+        if self.version != 3:
+            raise SourceMapVersionError(self.version)
+
+        self.sources: List[str] = source_map["sources"]
+
+        if "mapping" in source_map:
+            # For backwards compat until go-algorand has the change then we should remove
+            self.mappings: str = source_map["mapping"]
+        else:
+            self.mappings: str = source_map["mappings"]
+
+        pc_list = [
+            _decode_int_value(raw_val) for raw_val in self.mappings.split(";")
+        ]
+
+        # Initialize with 0,0 for pc/line
```

Comment:
@barnjamin Minor:  Comment appears to be stale because the `0,0` mapping no longer exists.

```suggestion
```